weaver/wps_restapi/providers/providers.py

```diff
@@ -10,44 +11,59 @@
 from weaver.exceptions import ServiceNotFound, log_unhandled_exceptions
 from weaver.formats import OUTPUT_FORMAT_JSON
 from weaver.owsexceptions import OWSMissingParameterValue, OWSNotImplemented
-from weaver.processes.types import PROCESS_WPS
+from weaver.processes.types import PROCESS_WPS_REMOTE
 from weaver.store.base import StoreServices
-from weaver.utils import get_any_id, get_settings
+from weaver.utils import get_any_id, get_settings, request_extra
 from weaver.warning import NonBreakingExceptionWarning
 from weaver.wps_restapi import swagger_definitions as sd
 from weaver.wps_restapi.utils import get_wps_restapi_base_url
 
 LOGGER = logging.getLogger(__name__)
+if TYPE_CHECKING:
+    from typing import List
 
+    from pyramid.request import Request
 
-@sd.providers_service.get(tags=[sd.TAG_PROVIDERS], renderer=OUTPUT_FORMAT_JSON,
-                          schema=sd.GetProviders(), response_schemas=sd.get_providers_list_responses)
-@log_unhandled_exceptions(logger=LOGGER, message=sd.InternalServerErrorGetProvidersListResponse.description)
-def get_providers(request):
+
+def get_provider_services(request):
+    # type: (Request) -> List[Service]
     """
-    Lists registered providers.
+    Obtain the list of remote provider services.
     """
-    store = get_db(request).get_store(StoreServices)
+    settings = get_settings(request)
+    store = get_db(settings).get_store(StoreServices)
     providers = []
-
     for service in store.list_services():
+        # pre-check service location
+        # status can be 500 because of missing query params, but faster skip of invalid references
+        # this avoids long pending connexions that never resolve because of down server
         try:
-            if service.type.lower() != "wps":
+            resp = request_extra("head", service.url, timeout=1, settings=settings)
+            if resp.status_code == 404:
+                LOGGER.warning("Skipping unresponsive service (%s) [%s]", service.name, service.url)
```

Comment:
Missing status 500 handling here?

Reply:
I'll add a note of special use case.
It is purposely ignored because missing `request`, `service`, etc. WPS parameters cause 500 by many services, but at least it 'answered'.